### DISCUSSION TITLE
How to place a image in center?

### DISCUSSION THREAD

**Discussion by Serenade600** [marp-team/marp#430] [Q&A] [answered]
I want to place a image centered. So I tried the following code in the particular slide page:
```
---

# Reactor

<style scoped>
img {justify-content: center}
</style>

![](./img/maze.png)

---
```

But the result I got is not centered. I screenshot this page, so please ignore the black border.

![image](https://user-images.githubusercontent.com/40297752/232661351-64323fca-517b-4053-8f0a-c66724cb4530.png)

Is there something I set wrong? I'am using marp in vs code.

**Comment by Serenade600** [comment]
I see here https://github.com/orgs/marp-team/discussions/283#discussioncomment-2393104. It said that only `background` syntax is used for figure. So I changed the previous code to 
```
<style scoped>
img {background-position: center}
</style>
```
Also didn't work. I also changed the element `img` to `figure`. Still didn't work. 
I'm a newbie about marp and html. Just attracted by the awesome feature by Marp. Hope this question doesn't offend anyone.

**Reply by yhatt** [reply]
#283 can be only applied for the background image `![bg]()`, [a unique feature of Marp/Marpit](https://marpit.marp.app/image-syntax?id=slide-backgrounds). You are using [an inline image syntax for general Markdown `![]()`](https://www.markdownguide.org/basic-syntax/#images-1), _that is not a background image_. Thus, `background-xxxxxxx` CSS property cannot apply in this case.

**Comment by yhatt** [comment, marked as answer]
```markdown
![](./img/maze.png)
```

This Markdown renders a following HTML:

```html
<p>
  <img src="./img/maze.png" />
</p>
```

`<p>` tag (paragraph) is aligning its contents to left (`text-align: left`). Thus, using a following style can align contents into center.

```html
<style scoped>
p { text-align: center; }
</style>
```

See also https://marp.app/blog/the-story-of-marp-next#image for more flexible way. It can define `center` keyword for the image. (**WARNING**: This way is abusing an alternative text of the image)
